src/main/java/core/mate/academy/service/MachineServiceImpl.java

```diff
@@ -1,7 +1,41 @@
 package core.mate.academy.service;
 
+import core.mate.academy.model.Bulldozer;
+import core.mate.academy.model.BulldozerProducer;
+import core.mate.academy.model.Excavator;
+import core.mate.academy.model.ExcavatorProducer;
+import core.mate.academy.model.Machine;
+import core.mate.academy.model.Truck;
+import core.mate.academy.model.TruckProducer;
+import java.util.List;
+
 /**
  * Your implementation of MachineService.
  */
-public class MachineServiceImpl {
+public class MachineServiceImpl<T extends Machine> implements MachineService<T> {
+    @Override
+    public List<T> getAll(Class<? extends T> type) {
+        if (type == Bulldozer.class) {
+            return (List<T>) new BulldozerProducer().get();
+        } else if (type == Excavator.class) {
+            return (List<T>) new ExcavatorProducer().get();
+        } else if (type == Truck.class) {
+            return (List<T>) new TruckProducer().get();
+        }
+        return List.of();
```

Comment:
Consider creating a local variable for the result of the `get()` method call. This can improve readability and makes it easier to handle the list before returning it, as suggested in the checklist.